assets/javascripts/lib/ajax.coffee

```diff
@@ -6,18 +6,32 @@ MIME_TYPES =
   applyDefaults(options)
   serializeData(options)
 
-  xhr = new XMLHttpRequest()
-  xhr.open(options.type, options.url, options.async)
-
-  applyCallbacks(xhr, options)
-  applyHeaders(xhr, options)
-
-  xhr.send(options.data)
-
-  if options.async
-    abort: abort.bind(undefined, xhr)
-  else
-    parseResponse(xhr, options)
+  abortController = new AbortController()
+
+  timer = setTimeout =>
+    abortController.abort()
+  , options.timeout * 1000
+
+  fetch(
+    options.url,
+    headers: processHeaders(options)
+    method: options.type
+    contentType: options.dataType
+    signal: abortController.signal
+  ).then((response) ->
+    if options.dataType == 'json'
+      response.json()
+    else
+      response.text()
```

Comment:
I think `response.ok` and `response.status` (`200 <= xhr.status < 300`) need to be checked.